arch/common/sw_isr_common.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sw_isr_table.h>
+#include <arch/cpu.h>
+
+/*
+ * Common code for arches that use software ISR tables (CONFIG_GEN_ISR_TABLES)
+ */
+
+#ifdef CONFIG_DYNAMIC_INTERRUPTS
+void z_isr_install(unsigned int irq, void (*routine)(void *), void *param)
+{
+	int table_idx = irq - CONFIG_GEN_IRQ_START_VECTOR;
+	int key = irq_lock();
```

Comment:
Don't see why you need to lock this.  If the user is in the process of installing a handler for an interrupt that is unmasked, then they have already lost the game.

Along the same lines, it couldn't hurt to add an __ASSERT(!irq_is_enabled(irq), "...") here.

Reply:
done.